src/assessments/seo/KeywordDensityAssessment.js

```diff
@@ -68,20 +127,29 @@ class KeywordDensityAssessment extends Assessment {
 	 * @returns {AssessmentResult} The result of the assessment.
 	 */
 	getResult( paper, researcher, i18n ) {
-		const assessmentResult = new AssessmentResult();
+		// Get the environment variable.
+		this._hasMorphologicalForms = researcher.getData( "morphology" ) !== false && paper.getLocale() === "en_EN";
 
 		this._keywordCount = researcher.getResearch( "keywordCount" );
+		const keyphraseLength = this._keywordCount.length;
+
+		this.setBoundaries( paper.getText(), keyphraseLength );
+
+		const assessmentResult = new AssessmentResult();
 
 		this._keywordDensity = researcher.getResearch( "getKeywordDensity" );
 
-		/*
-		 * Use other boundaries when taking morphology into account,
-		 * since multiple keyword forms can be matched.
-		 */
-		this._hasMorphologyData = researcher.getData( "morphology" ) !== false;
-		this._locale = paper.getLocale();
+		if ( process.env.YOAST_RECALIBRATION === "enabled" ) {
+			this._keywordDensity = this._keywordDensity * keyphraseLengthFactor( keyphraseLength );
```

Comment:
Small suggestions: I'd move this line down in the if statement below. That way, all recalibration functionality in the `getResult` function is within a single block, which makes it a bit more easy to see the difference between non-recalibration/recalibration.

Reply:
Good one 😄